lib/core/src/refund/chain.rs

```diff
@@ -0,0 +1,273 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use boltz_client::{
+    boltz::{BOLTZ_MAINNET_URL_V2, BOLTZ_TESTNET_URL_V2},
+    network::electrum::ElectrumConfig,
+    util::secrets::Preimage,
+    Amount, BtcSwapTx,
+};
+use log::info;
+use tokio::sync::Mutex;
+
+use crate::{
+    chain::{bitcoin::BitcoinChainService, liquid::LiquidChainService},
+    error::{PaymentError, SdkResult},
+    prelude::{ChainSwap, Config, Direction, LiquidNetwork, SwapScriptV2},
+    swapper::BoltzSwapper,
+};
+
+use super::{
+    common::{self, get_lbtc_fee_rate_sats_per_vbyte},
+    RefundHandler,
+};
+
+pub(crate) struct ChainRefundHandler<SwapperType: ?Sized> {
+    config: Config,
+    swapper: Option<Arc<SwapperType>>,
+    liquid_chain_service: Arc<Mutex<dyn LiquidChainService>>,
+    bitcoin_chain_service: Arc<Mutex<dyn BitcoinChainService>>,
+}
+
+impl ChainRefundHandler<BoltzSwapper> {
+    pub(crate) fn new(
+        config: Config,
+        swapper: Arc<BoltzSwapper>,
+        liquid_chain_service: Arc<Mutex<dyn LiquidChainService>>,
+        bitcoin_chain_service: Arc<Mutex<dyn BitcoinChainService>>,
+    ) -> Self {
+        Self {
+            config,
+            swapper: Some(swapper),
+            liquid_chain_service,
+            bitcoin_chain_service,
+        }
+    }
+
+    fn boltz_url(&self) -> &str {
+        self.swapper
+            .as_ref()
+            .map(|swapper| swapper.boltz_url())
+            .unwrap_or(match self.config.network {
+                LiquidNetwork::Mainnet => BOLTZ_MAINNET_URL_V2,
+                LiquidNetwork::Testnet => BOLTZ_TESTNET_URL_V2,
+            })
+    }
+
+    fn bitcoin_electrum_config(&self) -> ElectrumConfig {
+        self.swapper
+            .as_ref()
+            .map(|swapper| swapper.bitcoin_electrum_config().clone())
```

Comment:
If this is to be used in a refund handler without a swapper, then maybe it makes sense to de-couple `bitcoin_electrum_config()` from the swapper and perhaps expose it as a function of the main `Config` instead.

Otherwise we have the initialization code duplicated: once in the swapper, once here (as fallback if no swapper present). IMO this could be consolidated.